tests/account_change_password_test.go

```diff
@@ -0,0 +1,46 @@
+package tests
+
+import (
+	"testing"
+
+	"encoding/json"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestChangePassword(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	newPassword := "new_superuser"
+	// sytest: After changing password, can't log in with old password
+	t.Run("After changing password, can't log in with old password", func(t *testing.T) {
+		CreateDummyUser(t, unauthedClient, "cant_login_with_old_password")
+		res := unauthedClient.DoFunc(t, "POST", []string{"_matrix", "client", "r0", "account", "password"}, client.WithRawBody(json.RawMessage(`{
```

Comment:
I wouldn't use `client.WithRawBody` when you are just sending JSON. Use `client.WithJSONBody` instead, which accepts an `interface{}`. There's a few reasons for this:
 - It's easier to add variables to the JSON body, things like `newPassword` here.
 - It's easier to read and less error prone, as the compiler will complain if you miss a `,` but this isn't true for the raw form.
 - It will encode special characters correctly so it remains valid JSON.

```go
client.WithJSONBody(map[string]interface{} {
    "auth": map[string]interface{}{
        "type": "m.login.password",
        "user": "@cant_login_with_old_password:hs1",
        "password": "superuser",
    },
    "new_password": newPassword,
})

Reply:
Done the change!